src/Microsoft.DotNet.Interactive.Jupyter/Http/JupyterHttpConnection.cs

```diff
@@ -60,6 +74,29 @@ public void Dispose()
         _disposables.Dispose();
     }
 
+    public async Task<string[]> GetKernelSpecsAsync()
+    {
+        if (_availableKernels == null)
+        {
+            HttpResponseMessage response = await SendWebRequestAsync(
+                apiPath: "api/kernelspecs",
+                body: null,
+                contentType: "application/json",
+                method: HttpMethod.Get
+            );
+
+            if (!response.IsSuccessStatusCode)
+            {
+                return null;
```

Comment:
```suggestion
                return Array.Empty<string>();
```

It's good to avoid expecting that `null` will have some special meaning distinct from an empty sequence.